### DISCUSSION TITLE
Questions regarding: WaynesGrid, additional GUI controls, JSON

### DISCUSSION THREAD

**Discussion by adakordev** [twinbasic/twinbasic#664] [Q&A]
Hey all. Some questions:

1.)
Could someone please point me to where I can find examples on how to populate the GUI WaynesGrid rows/ columns from a list of values (in an efficient manner)?
Could this example code be added to Sample 6. CustomControls?

2.)
taken from: twinBASIC 12-MONTH ROADMAP #335

FORMS GUI SUPPORT (PART ONE) ☑ (STARTED)
EXPECTED RELEASE DATE: 10th January 2022

Does this mean additional GUI work will be released soon? 
I would really like to have some additional controls like checkboxes, dropdowns, being able to use a picture/ logo.

3.)
JSON support should not be far away as it is already used under the hood for the custom controls? As far as I remember only the syntax has to be decided?

**Comment by WaynePhillipsEA** [comment]
1) they'll be some updates to the CustomControls mid-next week, I'll put a little sample together for you.

2) Sorry, as usual, we're slightly behind.  The new package manager was needed in order to aid in the deployment of CustomControls going forward, which took out a good chunk of last weeks development time.

3) Yes, the work is largely done.  We just need to discuss how to incorporate it, especially considering things like serialisation of UDTs and objects.

**Comment by WaynePhillipsEA** [comment]
Here's an example project that utilizes WaynesGrid, using the new GetCellText event that I've just added (in CustomControlsPackage v0.0.2.0).
 
[FilePropertyViewer.zip](https://github.com/WaynePhillipsEA/twinbasic/files/7924246/FilePropertyViewer.zip)

<img width="436" alt="tbFilePropertyViewerForm" src="https://user-images.githubusercontent.com/25738109/150755672-e0ae6ec0-da1a-4719-9836-515a055e2993.png">

The main problem at the moment is the lack of scrollbars, but that will be added soon.

**Reply by WaynePhillipsEA** [reply]
Sorry, you will have to wait a while longer.  This is not in scope for inclusion in the initial v1 release, and as such it's considered low priority.

**Reply by wqweto** [reply]
> Wait for 3 Years

This discussion/issue has to be closed as all points are either complete, wontfix (JSON) or irrelevant (WaynesGrid, WTF?).

Please if you don't have anything constructive to add, don't post snarky remarks and/or try to vex the people who try to support the project get completed.

I'll personally ask/point admins to take actions the next time snark is detected in any of your posts here.
